### PR TITLE
Feature/allocators

### PR DESCRIPTION
**Pull request by koparasy**
Setups a better and more clean interface with umpire allocators.

### CODE REVIEW

File: examples/main.cpp [hunk 1]
```diff
@@ -311,28 +409,53 @@ int main(int argc, char **argv)
...
   // -------------------------------------------------------------------------
-  AMSSetupAllocator(AMSResourceType::HOST);
-  if (use_device) {
-    AMSSetupAllocator(AMSResourceType::DEVICE);
-    AMSSetupAllocator(AMSResourceType::PINNED);
-    AMSSetDefaultAllocator(AMSResourceType::DEVICE);
-  } else {
-    AMSSetDefaultAllocator(AMSResourceType::HOST);
-  }
+  //  AMSSetupAllocator(AMSResourceType::HOST);
```
Comment (ggeorgakoudis) [inline examples/main.cpp]: Is this commented-out code needed?

File: examples/main.cpp [hunk 2]
```diff
@@ -699,6 +821,7 @@ int main(int argc, char **argv)
     }
     CALIPER(CALI_MARK_END("Cycle");)
     MPI_CALL(MPI_Barrier(MPI_COMM_WORLD));
+    printMemory(allocator_names);
```
Comment (ggeorgakoudis) [inline examples/main.cpp:809]: Is printing always memory info intentional? Alternative is to guard it only for debugging

**Review by ggeorgakoudis — APPROVED** [review]
LGTM, fix minor comment

File: examples/main.cpp (hunk 2 above)
Comment (koparasy) [reply examples/main.cpp:809]: We currently use the example code as a very exhaustive test. I think we should always print out the memory consumption of all of our allocators.

File: examples/main.cpp (hunk 1 above)
Comment (koparasy) [reply examples/main.cpp]: Fixed in latest commit.

**Pull request merged**